wallet/src/de/schildbach/wallet/ui/send/SendCoinsFragment.java

```diff
@@ -1221,501 +586,215 @@ protected void onFailure(Exception exception) {
-        SendRequest sendRequest = createSendRequest(paymentIntent, RequestType.REGULAR_PAYMENT, false, false);
-        wallet.completeTx(sendRequest);
-        dryrunReqularPaymentFee = sendRequest.tx.getFee();
-    }
-
-    private enum RequestType {
+    private SendRequest createSendRequest(PaymentIntent paymentIntent, boolean signInputs, boolean forceEnsureMinRequiredFee) {
 
-        INSTANT_SEND_AUTO_LOCK,
-        INSTANT_SEND,
-        INSTANT_SEND_LLMQ,
-        REGULAR_PAYMENT;
-
-        public static RequestType from(SendRequest sendRequest) {
-            if (sendRequest.coinSelector == ZeroConfCoinSelector.get()) {
-                return REGULAR_PAYMENT;
-            } else if (sendRequest.coinSelector == InstantXCoinSelector.get()) {
-                return sendRequest.useInstantSend ? INSTANT_SEND : INSTANT_SEND_AUTO_LOCK;
-            } else {
-                throw new IllegalArgumentException();
-            }
-        }
-    }
-
-    private SendRequest createSendRequest(PaymentIntent paymentIntent, RequestType requestType, boolean signInputs, boolean forceEnsureMinRequiredFee) {
-        if (fees == null) {
-            throw new IllegalStateException();
-        }
+        final Wallet wallet = viewModel.wallet;
         boolean llmqInstantSendActive = wallet.getContext().sporkManager.isSporkActive(SporkManager.SPORK_20_INSTANTSEND_LLMQ_BASED);
 
-        paymentIntent.setInstantX(requestType == RequestType.INSTANT_SEND); //to make sure the correct instance of Transaction class is used in toSendRequest() method
+        paymentIntent.setInstantX(false); //to make sure the correct instance of Transaction class is used in toSendRequest() method
         final SendRequest sendRequest = paymentIntent.toSendRequest();
-        switch (requestType) {
-            case INSTANT_SEND_LLMQ:
-            case INSTANT_SEND_AUTO_LOCK: {
-                sendRequest.coinSelector = llmqInstantSendActive ? ZeroConfCoinSelector.get() : InstantXCoinSelector.get();
-                sendRequest.useInstantSend = false;
-                sendRequest.feePerKb = fees.get(feeCategory);
-                break;
-            }
-            case INSTANT_SEND: {
-                sendRequest.coinSelector = InstantXCoinSelector.get();
-                sendRequest.useInstantSend = true;
-                sendRequest.feePerKb = TransactionLockRequest.MIN_FEE;
-                break;
-            }
-            case REGULAR_PAYMENT: {
-                sendRequest.coinSelector = ZeroConfCoinSelector.get();
-                sendRequest.useInstantSend = false;
-                sendRequest.feePerKb = fees.get(feeCategory);
-                break;
-            }
-        }
-
-        sendRequest.ensureMinRequiredFee = forceEnsureMinRequiredFee ? true : sendRequest.useInstantSend;
+        sendRequest.coinSelector = llmqInstantSendActive ? ZeroConfCoinSelector.get() : InstantXCoinSelector.get();
```

Comment:
if llmqInstantSendActive is false, the `ZeroConfCoinSelector` can still be used.

We shouldn't use `InstantXCoinSelector` anymore

Reply:
There are no rules that help qualify a transaction for instantsend anymore, except that if the transactions are chained then there has to be less than 24 in the chain.  In this case the chain is a list set of transactions that are spending each others outputs, but none of them are confirmed in a block yet.  I am not sure if we can determine this easily.

Reply:
Thanks for the explanation!